zerver/lib/user_groups.py

```diff
@@ -384,7 +384,7 @@ def get_role_based_system_groups_dict(realm: Realm) -> Dict[str, UserGroup]:
 
 def set_defaults_for_group_settings(
     user_group: UserGroup,
-    group_settings_map: Mapping[str, UserGroup],
+    group_settings_map: Mapping[str, Union[str, UserGroup]],
```

Comment:
I think we can just receive a list of strings here instead of `Mapping` which will contain the name of the settings which are already set. I guess would be better to do it as a prep commit.